src/runner/kafka.ts

```diff
@@ -36,66 +51,151 @@ class KafkaRunner extends BaseRunner implements IRunner {
     this.registry = registry;
     this.logger = logger;
     this.pool = pool;
+    this.paused = false;
 
-    this.consumer = new Kafka.GroupConsumer({
-      groupId: config.kafkaGroupId,
-      clientId: config.clientId,
-      connectionString: config.kafkaConnection,
-      // @ts-ignore
-      codec: config.kafkaCodec,
-      logger: {
-        logLevel: config.logLevel,
+    this.consumer = new Kafka.KafkaConsumer(
+      {
+        'bootstrap.servers': this.config.bootstrapServers,
+        ...(this.config.consumer?.global ?? {}),
       },
+      this.config.consumer?.topic ?? {}
+    );
+
+    this.adminClient = AdminClient.create({
+      'bootstrap.servers': this.config.bootstrapServers,
+      ...this.config.admin,
     });
   }
 
-  receive = async (messages: any[], topic: string, partition: string) => {
-    for (const m of messages) {
-      // eslint-disable-line no-restricted-syntax
-      let params: any = {};
-      try {
-        // commit offset
-        params = JSON.parse(m.message.value.toString('utf8'));
-        const context = getContext(params);
-
-        this.registry.events.emit('runner_receive', topic, params, context);
-        await this.consumer.commitOffset({ topic, partition, offset: m.offset, metadata: 'optional' }); // eslint-disable-line
-        const task = this.registry.getTask(topic);
-        if (!task) {
-          this.logger.error(`Unknown Task ${topic}`);
-          continue;
+  receive = async (message: Message) => {
+    const { topic } = message;
+    try {
+      const parsed = {
+        ...message,
+        value: message.value?.toString(),
+      };
+      this.registry.events.emit('runner_receive', topic, parsed, {
+        ...message,
+        start: getDuration(),
+      });
+      const task = this.registry.getTask(topic);
+      if (!task) {
+        this.logger.error(`Unknown Task ${topic}`);
+        this.consumer.commitMessage(message);
+        return;
+      }
+      if (!this.config.waitToCommit) {
+        this.consumer.commitMessage(message);
+      }
+      this.logger.debug('Start subscribe', topic, message);
+      await task.subscribe(message);
+      if (this.config.waitToCommit) {
+        this.consumer.commitMessage(message);
```

Comment:
This isn't really what I had in mind for controlling commits - this loses a lot of potential flexibility in tasks to manage this. An example may be a task that does some critical work that must happen once then sends an analytics event that is ok to fail sporadically - it'd be more ideal for the task to be able to commit after the critical work and reduce the risk of hitting the consume timeout.